### ISSUE TITLE
Subglosses are not included

### ISSUE THREAD

**Issue by gd4c** [tatuylonen/wiktextract#160]
For example, https://en.wiktionary.org/wiki/dress senses 6 and 10 under transitive

**Comment by gd4c** [comment]
Also, thanks for the service 👍.

You probably do test the extracted data but maybe using the data to recreate the Wiktionary static page and compare to see if anything is missing?

**Comment by kristian-clausal** [comment]
> You probably do test the extracted data but maybe using the data to recreate the Wiktionary static page and compare to see if anything is missing?

Unfortunately, the parsing is strictly one way only because it's 'lossy'. Wiktionary is human-written, and not generated, so there is nothing to *generate with* so to speak; there's no underlying system or script that we could use to recreate Wiktionary pages based on the data we've extracted from them. I suspect what you're asking is a **really**, genuinely hard thing to do, probably impossible.

**Comment by gd4c** [comment]
@kristian-clausal My understanding is that, even though Wiktionary is human written, information is "slotted" in a standardized pattern. If the wiki can be parsed into predictable html, machine-readable Wiktextract output can be used to generate plain text to be compared with the text of the html pages.

Maybe it is my inexperience, but I don't think it is that difficult. Dictionaries [have already been made](https://github.com/Vuizur/Wiktionary-Dictionaries) from wiktextract data.

**Comment by kristian-clausal** [comment]
For testing purposes, generating a dictionary is insufficient. If you want to see whether something is missing, you'd have to recreate the original wiktionary page, which can't be done, or your tests are just human eyeballs checking for errors and missing things, which is what we're basically already doing.

> Does it have to be lossy or are you describing the current state of the extraction code?

I would say it is impossible to make a parser that can extract data from Wiktionary that isn't 'lossy'. The alternative to that is to keep enough data to recreate the page... Which is ultimately just keeping a copy of the original page, because there are so many ways to recreate the page that you can't feasibly 'unscramble the egg'.

**Comment by gd4c** [comment]
@kristian-clausal

>Which is ultimately just keeping a copy of the original page
But the difference is that it is in machine-readable format, as this project does.

Thanks for the explanation.

**Comment by kristian-clausal** [comment]
I've *finally* got working a remixed implementation of the sense gloss list parsing code that is purely recursive. Because it is recursive, it can go arbitrarily deep in a gloss list and generated glosses at the very end nodes, using data and glosses from higher up nodes (superglosses) as part of the more specific ones.

Along the way, I found a few bugs that drove me almost crazy, but I think I squished the most pertinent one.

I know I am tempting fate by pushing this to kaikki.org on a Thursday afternoon, but the only real consequence is that kaikki wouldn't update (that is, normal behavior when nothing changes) and I keep on getting annoying log emails reminding me that it's failed yet again over the weekend.
